### PR TITLE
modify docker file to run profiling

### PR DESCRIPTION
**Pull request by goncer**
Ready to review. 
export this right before running the node to have profiling.
`LD_PRELOAD=/mangata/libbytehound.so`

### CODE REVIEW

File: devops/dockerfiles/node/Dockerfile [hunk 1]
```diff
@@ -3,23 +3,36 @@ LABEL description="Compiles all workspace artifacts"
 WORKDIR /mangata
 COPY . /mangata
 
+#Export memory Profile env. variables.
+RUN export MEMORY_PROFILER_LOG=info && export MEMORY_PROFILER_LOGFILE=profiling_%e_%t.log && export MEMORY_PROFILER_OUTPUT=profiling_%e_%t.dat && export MEMORY_PROFILER_CULL_TEMPORARY_ALLOCATIONS=1
```
Comment (mateuszaaa) [inline devops/dockerfiles/node/Dockerfile:7]: please built in dockerfile `ENV` command 
```
ENV MEMORY_PROFILER_LOG=info
ENV MEMORY_PROFILER_LOGFILE=profiling_%e_%t.log
...
```

File: devops/dockerfiles/node/Dockerfile [hunk 2]
```diff
@@ -3,23 +3,36 @@ LABEL description="Compiles all workspace artifacts"
...
 
 RUN rustup install nightly-2021-10-19
 RUN rustup default nightly-2021-10-19
 RUN rustup target add wasm32-unknown-unknown --toolchain nightly-2021-10-19
+#Get Mem profiler binaries
+
 RUN cargo install cargo2junit
 
 FROM env_builder as builder
+RUN mkdir memory_profiler && cd memory_profiler && git clone https://github.com/koute/memory-profiler && cd memory-profiler && git checkout  0.8.0  &&  cargo build --release -p bytehound-preload && cargo build --release -p bytehound-cli
```
Comment (mateuszaaa) [inline devops/dockerfiles/node/Dockerfile:23]: maybe it works but actully i'm not happy how its done;/ we still build profiler binaries on image built on top of 
`liuchong/rustup:stable` where we run these in another docker image  ;/ Ideally we should use just debian image and install rustup & cargo manually. But if it works (and it seems it does) we can just create ticker for later

File: devops/parachain/docker-compose.yml [hunk 3]
```diff
@@ -72,6 +72,7 @@ services:
     image: ${PARACHAIN_DOCKER_IMAGE:-mangatasolutions/mangata-node:latest}
     environment:
       - ${RUST_LOG:-info}
+      - LD_PRELOAD=${LD_PRELOAD}
```
Comment (mateuszaaa) [inline devops/parachain/docker-compose.yml:75]: maybe `${LD_PRELOAD:-""}` to stick to single convention ?

**Review by mateuszaaa — CHANGES_REQUESTED** [review]
good job

File: devops/dockerfiles/node/Dockerfile (hunk 1 above)
Comment (goncer) [reply devops/dockerfiles/node/Dockerfile:7]: +1

File: devops/parachain/docker-compose.yml (hunk 3 above)
Comment (goncer) [reply devops/parachain/docker-compose.yml:75]: lets see if this works :)

Comment (goncer) [reply devops/parachain/docker-compose.yml:75]: It does not, when running a command like this,

`export MANGATA_NODE_VERSION=442771a8aff53309db52f472fea8ae593c71b76a.tar; docker-compose -f ./docker-compose.yml down ; LD_PRELOAD=/mangata/libbytehound.so  docker-compose -f ./docker-compose.yml up`
It does not activate the profiler. Leaving it as it was before.

**Review by mateuszaaa — APPROVED** [review]
LGTM

**Comment by goncer** [comment]
@gleb-urvanov  this can be merged!

**Pull request merged**